### ISSUE TITLE
Calling extern functions with generic CFuncPtr fails at runtime 

### ISSUE THREAD

**Issue by jokade** [scala-native/scala-native#1627]
With **SN-0.4.0-M2**, calling an extern function that has a `CFuncPtr` parameter (instead of the more specific `CFuncPtrN[]`) fails at runtime. Example:

```C
typedef int (*Callback0) (void);

int exec(Callback0 f) {
  return f();
}
```

```scala
@extern
object ext {
  @name("exec")
  def exec0(f: CFuncPtr0[Int]): Int = extern
  def exec(f: CFuncPtr): Int = extern
}

val cb = new CFuncPtr0[Int] { override def apply(): Int = 42 }

// works as expected
ext.exec0(cb)

// fails at runtime
ext.exec(cb)
```

**Comment by jokade** [comment]
Any update on this? Will this ever work (i.e. passing CFunPtrN where CFuncPtr is expected)? Or do we need to cast to CFuncPtrN to Ptr[Byte] (via #1845).

It would be helpful to know if this is expected to work with SN-0.4.0 final, since it is a serious impediment for interop with libraries having a generic function for callback registration.

**Comment by WojciechMazur** [comment]
@jokade It was a bug, I've managed to fix it. I'll publish PR later and it should be included in SN-0.4.0
